web/src/main/java/com/microsoft/applicationinsights/web/internal/correlation/tracecontext/Traceparent.java

```diff
@@ -0,0 +1,185 @@
+    }
+
+    public int getTraceFlags() {
+        return traceFlags;
+    }
+
+    public String getSpanId() {
+        return spanId;
+    }
+
+    /**
+     * Validates the given input based on W3C specifications.
+     */
+    private static void validate(int version, String traceId, String spanId, int traceFlags)
+        throws IllegalArgumentException {
+        if (version < 0 || version > 254) {
+            throw new IllegalArgumentException("version must be within range [0, 255)");
+        }
+        if (!isHex(traceId, 32)) {
+            throw new IllegalArgumentException("invalid traceId");
+        }
+        if (traceId.equals("00000000000000000000000000000000")) {
+            throw new IllegalArgumentException("invalid traceId");
+        }
+        if (!isHex(spanId, 16)) {
+            throw new IllegalArgumentException("invalid spanId");
+        }
+        if (spanId.equals("0000000000000000")) {
+            throw new IllegalArgumentException("invalid spanId");
+        }
+        if (traceFlags < 0 || traceFlags > 255) {
+            throw new IllegalArgumentException("traceFlags must be within range [0, 255]");
+        }
+    }
+
+    /**
+     * Converts the Traceparent object to header format Eg: 00-0af7651916cd43dd8448eb211c80319c-b7ad6b7169203331-01
+     *
+     * @return traceparent
+     */
+    @Override
+    public String toString() {
+        return String.format("%02x-%s-%s-%02x", version, traceId, spanId, traceFlags);
+    }
+
+    /**
+     * Helper method to create a random hexadecimal string of n bytes.
+     *
+     * @return n byte hexadecimal string
+     */
+    @VisibleForTesting
+    static String randomHex(int n) {
+        byte[] bytes = new byte[n];
+        ThreadLocalRandom.current().nextBytes(bytes);
+        StringBuilder sb = new StringBuilder();
+        for (byte b : bytes) {
+            sb.append(String.format("%02x", b));
+        }
+        return sb.toString();
+    }
+
+    /**
+     * Helper method to check if a given string of n bytes is hexadecimal
+     *
+     * @return boolean
+     */
+    private static boolean isHex(String s, int n) {
+        if (s == null || s.length() == 0) {
+            return false;
+        }
+        if (s.length() != n) {
+            return false;
+        }
+        for (int i = 0; i < n; i++) {
+            char c = s.charAt(i);
+            if ('0' <= c && c <= '9') {
+                continue;
+            }
+            if ('a' <= c && c <= 'f') {
+                continue;
+            }
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * converts traceparent from String to Traceparent object
+     *
+     * @return Traceparent
+     */
+    public static Traceparent fromString(String s) {
+        if (s == null || s.length() == 0) {
+            return null;
+        }
+        String[] arr = s.split("-");
+        if (arr.length != 4) {
+            return null;
+        }
+        if (!isHex(arr[0], 2)) {
```

Comment:
Depending on the usage here, you should consider replacing the `isHex` checks and instead assign the `Character.digit` calls to local variables, then check for `-1` since the `.digit` method is doing these checks already.